torch/onnx/experimental/fonnx/v1/export_options.py

```diff
@@ -0,0 +1,12 @@
+@dataclasses.dataclass
+class ExportOptions:
+    """Class for keeping track of ONNX exporter flags can perform."""
+
+    opset_version: Optional[int] = 17
+    """opset_version: Target ONNX opset version"""
+
+    dynamic_shape: Optional[bool]
+    """Defines shape information for input/output tensors.
+
+    When True, all inputs are considered dynamic.
+    Otherwise, they can be either dynamic or static, depending on FXGraphExtractor's implementation"""
```

Comment:
>    When True, all inputs are considered dynamic.
    Otherwise, they can be either dynamic or static, depending on FXGraphExtractor's implementation"""

Does it mean `dynamic_shape` overrides `FXGraphExtractor's implementation`, when `True`?
I feel this is tricky. When an option interacts with potentially another option. I feel it's unclear what user can expect with this flag.

Reply:
That is the option that we discussed during the last meeting. In that discussion, we agreed that, for v1 timeframe, a simple boolean would suffice to mark all inputs as dynamic or static.

In future refinements, users be able to do a fine-grained dynamic axes configuration. We realized the possible conflict between both `dynamic_shape` and `FXGraphExtractor`'s arguments and decided it was Ok to raise an exception for the user saying `raise RuntimeError("Either use 'dynamic_shape' or 'FXGraphExtrator''s dynamic axes, not both")`

In fact, we agreed on the same for both `verbose` and `diagnotics`. If user set `verbose=True`, internally we would set a `ExportDiagnostic` with basic settings. Or else, the user can specify their custom diagnostic setting.

Makes sense?

Reply:
updated as offline discussion. improved docstring to make clear this is a best effort type hint